src/main/java/com/example/demo/repository/BookRepository.java

```diff
@@ -3,12 +3,15 @@
 import com.example.demo.model.Book;
 import java.util.List;
 import java.util.Optional;
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Query;
+import org.springframework.data.repository.query.Param;
 
-public interface BookRepository {
+public interface BookRepository extends JpaRepository<Book, Long> {
 
-    List<Book> getAll();
+    @Query("SELECT b FROM Book b WHERE b.isDeleted = false")
+    List<Book> findAllActive();
 
-    Optional<Book> findBookById(Long id);
-
-    Book createBook(Book book);
+    @Query("SELECT b FROM Book b WHERE b.id = :id AND b.isDeleted = false")
+    Optional<Book> findActiveById(@Param("id") Long id);
```

Comment:
remove unused method

Reply:
The method were removed previously. If not sth went wrong with my PR